recipes/libssh2/all/patches/0002-openssl.patch

```diff
@@ -0,0 +1,29 @@
+diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
+index 296693a..e5d4dbb 100644
+--- a/src/CMakeLists.txt
++++ b/src/CMakeLists.txt
+@@ -67,12 +67,12 @@ if(CRYPTO_BACKEND STREQUAL "OpenSSL" OR NOT CRYPTO_BACKEND)
+ 
+   find_package(OpenSSL ${SPECIFIC_CRYPTO_REQUIREMENT})
+ 
+-  if(OPENSSL_FOUND)
++  if(OpenSSL_FOUND)
+     set(CRYPTO_BACKEND "OpenSSL")
+     set(CRYPTO_SOURCES openssl.c openssl.h)
+     list(APPEND PRIVATE_COMPILE_DEFINITIONS LIBSSH2_OPENSSL)
+-    list(APPEND PRIVATE_INCLUDE_DIRECTORIES ${OPENSSL_INCLUDE_DIR})
+-    list(APPEND LIBRARIES ${OPENSSL_LIBRARIES})
++    list(APPEND PRIVATE_INCLUDE_DIRECTORIES ${OpenSSL_INCLUDE_DIR})
++    list(APPEND LIBRARIES ${OpenSSL_LIBRARIES})
```

Comment:
This is weird, we're manually creating these variables in the openssl module:

https://github.com/conan-io/conan-center-index/blob/9f038df28e5a77044ac2230673b16465a6c20bbd/recipes/openssl/3.x.x/conanfile.py#L675-L677


https://github.com/conan-io/conan-center-index/blob/9f038df28e5a77044ac2230673b16465a6c20bbd/recipes/openssl/3.x.x/conanfile.py#L692-L694

Do you have an idea why this is not working?

Reply:
Yeah, but for some reason they don't show up in the generated cmake find scripts - probably only some other cmake generator configuration.

Reply:
Doesn't the generated `FindOpenSSL.cmake` module include a `conan-official-openssl-variables.cmake` file?

Reply:
It works fine for me without these patches, your openssl recipe is outdated I guess.